components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1620,7 +1621,7 @@ where
                 let mut exists = false;
                 if let Some(p) = util::find_peer_mut(&mut region, store_id) {
                     exists = true;
-                    if !p.get_is_learner() || p.get_id() != peer.get_id() {
+                    if !is_learner(p) || p.get_id() != peer.get_id() {
```

Comment:
Probably always update peer.

Reply:
We don't need to, the peer already exists, if the request is not a promote learner request then it will be rejected.

Reply:
What about demoting?

Reply:
Currently, demoting a voter to learner is not allowed, though https://github.com/tikv/tikv/pull/8401 will change this behavior.